chroma-agent/chroma_agent/lib/node_admin.py

```diff
@@ -1,36 +1,42 @@
 # Copyright (c) 2017 Intel Corporation. All rights reserved.
 # Use of this source code is governed by a MIT-style
 # license that can be found in the LICENSE file.
-
-
+import errno
 from jinja2 import Environment, PackageLoader
+from chroma_agent.lib.shell import AgentShell
+
 env = Environment(loader=PackageLoader('chroma_agent', 'templates'))
 
-from iml_common.lib.util import platform_info
-from chroma_agent.lib.shell import AgentShell
+NM_STOPPED_RC = 8  # network manager stopped
 
 
 def write_ifcfg(device, mac_address, ipv4_address, ipv4_netmask):
     ifcfg_tmpl = env.get_template('ifcfg-nic')
     ifcfg_path = "/etc/sysconfig/network-scripts/ifcfg-%s" % device
     with open(ifcfg_path, "w") as f:
-        f.write(ifcfg_tmpl.render(device = device, mac_address = mac_address,
-                                  ipv4_address = ipv4_address,
-                                  ipv4_netmask = ipv4_netmask))
+        f.write(ifcfg_tmpl.render(device=device,
+                                  mac_address=mac_address,
+                                  ipv4_address=ipv4_address,
+                                  ipv4_netmask=ipv4_netmask))
 
     return ifcfg_path
 
 
 def unmanage_network(device, mac_address):
-    """Rewrite the network configuration file to set NM_CONTROLLED="no"
-    TODO: This is destructive and overwrites the file loosing all settings.
-    This needs to be fixed up.
+    """
+    Rewrite the network configuration file to set NM_CONTROLLED="no"
+
+    TODO: This is destructive and overwrites the file clearing
+    previously configured settings, needs fixing.
     """
     ifcfg_path = write_ifcfg(device, mac_address, None, None)
 
-    if platform_info.distro_version >= 7.0:
+    if ifcfg_path:
         try:
             AgentShell.try_run(['nmcli', 'con', 'load', ifcfg_path])
+        except OSError as e:
+            if e.errno != errno.ENOENT:
+                raise e
         except AgentShell.CommandExecutionError as cee:
-            if cee.result.rc not in [127, 2, 8]:            # network manager may be uninstalled (127) stopped (8)
-                raise
+            if cee.result.rc != NM_STOPPED_RC:
```

Comment:
This should be:

` if cee.result.rc != errno.ENOEXEC:`

Reply:
ENOEXEC would mean that it couldn't be started correct?

Reply:
Macro: int ENOEXEC
Invalid executable file format. This condition is detected by the exec functions; see 26.5 Executing a File.

So I think this one is when the file exists but is not a valid executable format for the system/architecture.

_ENOENT_ should be the right one for the executable could not be found.

Reply:
Macro: int ENOEXEC
Invalid executable file format. This condition is detected by the exec functions; see 26.5 Executing a File.

I.e. Wrong executable format for the system/architecture (i.e. trying to run an Arm64 on x86_64).

Reply:
> Is NM really mapping exit codes to errnos?

Yes, see:

https://github.com/intel-hpdd/intel-manager-for-lustre/blob/3d8865f56e8702740b9a779e4e7c12614e397dfa/chroma-agent/chroma_agent/lib/node_admin.py#L10

Which is just an alias to `8`.

Reply:
Ah, I understand the meaning of your question now, yes they aren't using errno for their returns.

Reply:
So, then should we be using errno values when testing them?

Reply:
No, I think it's fine to have this as a module level constant.